app/infrastructure/broadcaster.py

```
@@ -0,0 +1,45 @@
+class MetricsBroadcaster:
+    def __init__(self):
+        self._lock = asyncio.Lock()
+        self._subscribers: set[asyncio.Queue] = set()
+
+    async def subscribe(self, max_queue_size: Optional[int] = 100) -> asyncio.Queue:
+        queue = asyncio.Queue(maxsize=max_queue_size)
+        async with self._lock:
+            self._subscribers.add(queue)
+        return queue
+
+    async def unsubscribe(self, queue: asyncio.Queue):
```

Ruff (0.12.2)

Missing return type annotation for public function `unsubscribe`

Add return type annotation: `None`

(ANN201)